src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -282,18 +282,26 @@ private void processResponseFileByHashcode(BatchInputUploadFileFilenameProcessor
     private void processBatchResponseFiles(BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor, ArmBatchResponses armBatchResponses) {
         armBatchResponses.getArmBatchResponseMap().values().forEach(
             armResponseBatchData -> {
-                if (nonNull(armResponseBatchData.getInvalidLineFileFilenameProcessor())
+                if ((CollectionUtils.isNotEmpty(armResponseBatchData.getInvalidLineFileFilenameProcessors())
+                    && (armResponseBatchData.getInvalidLineFileFilenameProcessors().size() == 1))
                     && (nonNull(armResponseBatchData.getCreateRecordFilenameProcessor())
                     || nonNull(armResponseBatchData.getArmResponseUploadFileRecord()))) {
 
                     preProcessResponseFilesActions(armResponseBatchData.getExternalObjectDirectoryId());
 
                     processInvalidLineFile(armResponseBatchData.getExternalObjectDirectoryId(),
-                                           armResponseBatchData.getInvalidLineFileFilenameProcessor(),
-                                           armResponseBatchData.getArmResponseInvalidLineRecord(),
+                                           armResponseBatchData.getInvalidLineFileFilenameProcessors().getFirst(),
+                                           armResponseBatchData.getArmResponseInvalidLineRecords().getFirst(),
```

Comment:
Should this process all entries in getArmResponseInvalidLineRecords() ?

Reply:
This checks that the getInvalidLineFileFilenameProcessors has size of 1 but not the getArmResponseInvalidLineRecords()

Reply:
There can only be one as the addition of the objects add the 2 types of objects at the same time. Its completely unnecessary to add another check for the size of the other list.

Reply:
As a general rule I think its better to protect against the edge cases, to ensure data integrity, and to allow for easier re-use. However, I'm happy if there is just a comment in both sections of code (This section and where the items get added) to ensure if we edit the code in the future we know we must add to both

Reply:
If we were to codify a second `size()` check, it's not clear if the second negative case would be reachable code (from a either a test or practical perspective), based on my reading of the calling code.

So on that basis I agree that a comment sounds reasonable, stating the invariant that _both_ `armResponseInvalidLineRecords` and `invalidLineFileFilenameProcessors` are expected to contain one (and only one) item.